analyzers/tests/SonarAnalyzer.Test/LiveVariableAnalysis/RoslynLiveVariableAnalysisTest.FlowCaptureOperation.cs

```diff
@@ -0,0 +1,354 @@
+        var context = CreateContextCS(code, additionalParameters: "string s1");
+        context.ValidateEntry(LiveIn("s1"), LiveOut("s1"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("s1"));
+        context.Validate(context.Cfg.Blocks[2], null);
+        context.Validate(context.Cfg.Blocks[3], null, LiveOut("s1"));
+        context.Validate(context.Cfg.Blocks[4], null, LiveOut("s1"));
+        context.Validate(context.Cfg.Blocks[5], null, LiveIn("s1"));
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_SwitchStatement()
+    {
+        const string code = """
+            string result = i switch
+            {
+                0 =>  stringParam,
+                1 => "Something",
+                _ => "Everything"
+            };
+            """;
+        var context = CreateContextCS(code, additionalParameters: "int i, string stringParam");
+        context.ValidateEntry(LiveIn("i", "stringParam"), LiveOut("i", "stringParam"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("i", "stringParam"), LiveOut("i", "stringParam")); // if i == 0
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("stringParam"), LiveOut("stringParam"));           // #0 = stringParam
+        context.Validate(context.Cfg.Blocks[3], null, LiveIn("i", "stringParam"), LiveOut("i", "stringParam")); // if i == 1
+        context.Validate(context.Cfg.Blocks[4], null, LiveIn("stringParam"), LiveOut("stringParam"));           // #0 = "Something"
+        context.Validate(context.Cfg.Blocks[5], null, LiveIn("i", "stringParam"), LiveOut("stringParam"));      // if i != 0 && i != 1
+        context.Validate(context.Cfg.Blocks[6], null, LiveIn("stringParam"), LiveOut("stringParam"));           // #0 = "Everything"
+        context.Validate(context.Cfg.Blocks[7], null);                                                          // to no destination block
+        context.Validate(context.Cfg.Blocks[8], null, LiveIn("stringParam"));                                   // result = #0
+        context.ValidateExit();
+    }
+
+    [TestMethod]
+    public void FlowCaptrure_ForEachCompundAssignment()
+    {
+        const string code = """
+            int sum = 0;
+            foreach (var i in list)
+            {
+                sum += i;
+            }
+            """;
+        var context = CreateContextCS(code, additionalParameters: "System.Collections.Generic.List<int> list");
+        context.ValidateEntry(LiveIn("list"), LiveOut("list"));
+        context.Validate(context.Cfg.Blocks[1], null, LiveIn("list"), LiveOut("list", "sum"));
+        context.Validate(context.Cfg.Blocks[2], null, LiveIn("list", "sum"), LiveOut("sum"));
+        context.Validate(context.Cfg.Blocks[3], null, LiveIn("sum"), LiveOut("sum"));
+        context.Validate(context.Cfg.Blocks[4], null, LiveIn("sum"), LiveOut("sum"));
+        context.Validate(context.Cfg.Blocks[5], null);
```

Comment:
Why doesn't have 5 LiveIn("list") when it captures it?

Reply:
Is it possible because you added the connections from other blocks to finally recently and I haven't rebased yet?

Reply:
Found it - I'm not dealing at all with these operations because I get only what is directly enclosed in local lifetime regions.

Reply:
Okay, so I tested by adding all the blocks in the analysis, and it still does not appear in the live in/live out, but now at least, it appears in the dictionary after the capture resolution.

What I saw is that the capture 0 points to the Invocation expression for the enumerator, and then capture 1 points to a Conversion operation with type IDisposable.

Reply:
Question is: Should I add all the blocks and not only the local lifetime ones?
Or add explicitly try catch region blocks too?

Reply:
I didn't appear. I'd say 
5 should have LiveIn, LiveOut
6 should have LiveIn
Add comments

Reply:
The problem is that you only process block that are directly inside a local lifetime region. If they are nested deeper, you don't find them.

What you need is this:
```
x.EnclosingRegion.EnclosingRegionOrSelf(ControlFlowRegionKind.LocalLifetime) is not null
```

Reply:
I expected some changes in the UTs after this :/